tur-on-device/awscli/build.sh

```diff
@@ -0,0 +1,152 @@
+		QYmVr3aIUes20i6meI3fuV36VFupwfrTKaL7VXnsrK2fq5cRvyJLNzXucg0WAjPF
+		RrAGLzY7nP1xeg1a0aeP+pdsqjqlPJom8OCWc1+6DWbg0jsC74WoesAqgBItODMB
+		rsal1y/q+bPzpsnWjzHV8+1/EtZmSc8ZUGSJOPkfC7hObnfkl18h+1QtKTjZme4d
+		H17gsBJr+opwJw/Zio2LMjQBOqlm3K1A4zFTh7wBC7He6KPQea1p2XAMgtvATtNe
+		YLZATHZKTJyiqA==
+		=vYOk
+		-----END PGP PUBLIC KEY BLOCK-----
+	EOF
+}
+
+_get_awscli_src_tarball() {
+	local tarball
+	local sig
+
+	_import_awscli_pgp_key
+
+	tarball="$(mktemp -p "$TERMUX_PKG_TMPDIR" "awscli.XXXXXX.tar.gz")"
+	sig="$(mktemp -p "$TERMUX_PKG_TMPDIR" "awscli.XXXXXX.sig")"
+
+	if [[ "${*}" =~ "--latest" ]]; then
+		curl -Lo "${tarball}" https://awscli.amazonaws.com/awscli.tar.gz
+		curl -Lo "${sig}" https://awscli.amazonaws.com/awscli.tar.gz.sig
+	else
+		curl -Lo "${tarball}" "${TERMUX_PKG_SRCURL}"
+		curl -Lo "${sig}" "${TERMUX_PKG_SRCURL}.sig"
+	fi
+
+	if ! gpg --verify "${sig}" "${tarball}"; then
+		rm -f "${sig}"
+		rm -f "${tarball}"
+		termux_error_exit "Error: failed to validate upstream source code"
+	fi
+
+	rm -f "${sig}"
+
+	echo "${tarball}"
+}
+
+termux_pkg_auto_update() {
+	local tarball
+	local version
+
+	tarball="$(_get_awscli_src_tarball --latest)"
+	version="$(tar tzf "${tarball}" | head -n 1 | grep -oP "\d+\.\d+\.\d+")"
+	rm -f "${tarball}"
+
+	if [ -z "${version}" ]; then
+		termux_error_exit "Error: failed to extract latest version"
+	fi
+
+	if [ "${TERMUX_PKG_VERSION}" != "${version}" ]; then
+		termux_pkg_upgrade_version "${version}" --skip-version-check
+	fi
+}
+
+termux_step_get_source() {
+	local tarball
+	tarball="$(_get_awscli_src_tarball)"
+	mkdir -p "${TERMUX_PKG_SRCDIR}"
+	tar --strip-components=1 -xf "${tarball}" -C "${TERMUX_PKG_SRCDIR}"
+	rm -f "${tarball}"
+}
+
+termux_step_pre_configure() {
+	if ! ${TERMUX_ON_DEVICE_BUILD}; then
+		termux_error_exit "This package doesn't support cross-compiling."
+	fi
+
+	# Unneeded dependency since we have python>=3.10
+	sed -i '/ruamel.yaml.clib/d' pyproject.toml
+	# Unsafe dependency
+	sed -i '/pip/d' requirements/bootstrap.txt
+
+	local awscrt_version
+	local cryptography_version
+	awscrt_version="$(grep -oP 'awscrt\K[=.<,>\d]+' pyproject.toml)"
+	cryptography_version="$(grep -oP 'cryptography\K[=.<,>\d]+' pyproject.toml)"
+
+	# We'll install these dependencies manually
+	sed -i '/awscrt/d' pyproject.toml
+	sed -i '/cryptography/d' pyproject.toml
+
+	pip3 install pip-tools
+	local requirements
+	requirements="$(mktemp -p . 'awscli-requirements.XXXXXX.txt')"
+	pip-compile --strip-extras --allow-unsafe --no-annotate -qo "${requirements}" \
+		requirements/download-deps/bootstrap.txt requirements/portable-exe-extras.txt pyproject.toml
+	pip3 install -r "${requirements}"
+	rm -f "${requirements}"
+
+	termux_setup_rust
+	pip3 install "cryptography${cryptography_version}"
+
+	termux_setup_cmake
+	pip3 install "awscrt${awscrt_version}"
+}
+
+termux_step_configure() {
+	./configure --prefix="${TERMUX_PREFIX}" --with-install-type=portable-exe
```

Comment:
It would be more suitable to install aws-cli-v2 as a python package, rather than a standalone binary. Some extra licenses should be added to this if we release a standalone binary as it will package so many third-party software.

Reply:
aws-cli-v2 is not available on pypi aws/aws-cli#4947

If we do not package as a standalone binary, then this package needs to have a dependency on python. In that case an update to python3 in termux-packages could break this package when python is updated. AWS does not always like to support the latest software, so building a portable binary would ensure other package updates do not break this package. aws/aws-cli#8342 for example

From a licensing perspective, would we need to include licenses from all of aws-cli-v2 dependencies along with this package? I would think as long as we package with the license aws distributes with their source code we should be good (I am not a lawyer though). If we package as a portable binary, we are not distributing any of the dependencies, but only the `aws` binary so I only see the need for the Apache-2.0 License here